### PR TITLE
Add compatibility tests for plugin arguments

### LINKED ISSUE #88027: Scheduler: Add compatibility tests for framework configuration
`pkg/scheduler/apis/config/testing` has compatibility tests for Policy v1.

We are lacking similar tests for other options of `scheduler.New`, such as:

- [x] `WithFrameworkPlugins`: This can be added right away to test that default plugins can be disabled, changed order, etc.
- [ ] `WithFrameworkPluginConfig`: This one is trickier, as it requires a new method in `Framework` to expose all configurations and a new method in `Plugin` so that they can say which args they used.

/sig scheduling
/help
/kind cleanup

### PR DESCRIPTION
**Pull request by Ye-Tian-Zero**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Add compatibility test for plugin arguments.

**Which issue(s) this PR fixes**:

Fixes #88027

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: pkg/scheduler/apis/config/testing/compatibility_test.go [hunk 1]
```diff
@@ -1794,3 +1795,284 @@ func TestPluginsConfigurationCompatibility(t *testing.T) {
 		})
 	}
 }
+
+func Test(t *testing.T) {
```
Comment (notpad) [inline pkg/scheduler/apis/config/testing/compatibility_test.go]: This seems to be duplicate of `TestAlgorithmProviderCompatibility`

File: pkg/scheduler/scheduler.go [hunk 2]
```diff
@@ -204,6 +205,16 @@ func WithExtenders(e ...schedulerapi.Extender) Option {
 	}
 }
 
+// FrameworkCapturer is used for registering a notify function in building framework.
+type FrameworkCapturer func(schedulerapi.KubeSchedulerProfile)
+
+// WithBuildFrameworkCapturer sets a notify function for getting buildframework details.
```
Comment (notpad) [inline pkg/scheduler/scheduler.go]: buildframework -> buildFramework

File: pkg/scheduler/apis/config/testing/compatibility_test.go (hunk 1 above)
Comment (alculquicondor) [inline pkg/scheduler/apis/config/testing/compatibility_test.go]: Algorithm provider is going away very soon. Remove this.

File: pkg/scheduler/apis/config/testing/compatibility_test.go [hunk 3]
```diff
@@ -1794,3 +1795,284 @@ func TestPluginsConfigurationCompatibility(t *testing.T) {
...
+			defProf := sched.Profiles["default-scheduler"]
+			gotPlugins := defProf.ListPlugins()
+			if diff := cmp.Diff(tc.wantPlugins, gotPlugins); diff != "" {
+				t.Errorf("unexpected plugins diff (-want, +got): %s", diff)
+			}
+		})
+	}
+}
+
+func TestPluginsArgumentsCompatibility(t *testing.T) {
```
Comment (alculquicondor) [inline pkg/scheduler/apis/config/testing/compatibility_test.go]: Make these tests part of  `TestPluginsConfigurationCompatibility`

File: pkg/scheduler/apis/config/testing/compatibility_test.go [hunk 4]
```diff
@@ -1794,3 +1795,284 @@ func TestPluginsConfigurationCompatibility(t *testing.T) {
...
+				informerFactory,
+				informerFactory.Core().V1().Pods(),
+				recorderFactory,
+				make(chan struct{}),
+				scheduler.WithProfiles(config.KubeSchedulerProfile{
+					SchedulerName: v1.DefaultSchedulerName,
+					PluginConfig:  tc.pluginConfig,
+				}),
+				scheduler.WithBuildFrameworkCapturer(func(p config.KubeSchedulerProfile) {
+					if diff := cmp.Diff(tc.pluginConfig, p.PluginConfig); diff != "" {
```
Comment (alculquicondor) [inline pkg/scheduler/apis/config/testing/compatibility_test.go]: Remember that args could be different for different profiles.

File: pkg/scheduler/apis/config/testing/compatibility_test.go (hunk 1 above)
Comment (skilxn-go) [reply pkg/scheduler/apis/config/testing/compatibility_test.go]: It's an error, I have removed this.

File: pkg/scheduler/apis/config/testing/compatibility_test.go (hunk 4 above)
Comment (skilxn-go) [reply pkg/scheduler/apis/config/testing/compatibility_test.go]: Sure, I will add other cases to cover this.

Comment (alculquicondor) [reply pkg/scheduler/apis/config/testing/compatibility_test.go]: not necessarily. I think you can cover most of it by merging these checks into `TestPluginsConfigurationCompatibility`

**Comment by alculquicondor** [comment]
/approve

You can squash now

**Comment by skilxn-go** [comment]
It's squashed

**Pull request merged**